### PR TITLE
Feature/335 initialize icp

### PR DESCRIPTION
**Pull request by thibaudteil**
* **Tickets addressed:** bsk-335
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
This module is needed to initialize the ICP process. It uses known spacecraft attitude/ephemeris information or previous ICP information to set the R,S,T values for ICP. 

## Verification
Unit test was added to test functionality

## Documentation
Documentation in rst form was added

## Future work
Improvements need to be added to use the filter solution and update which messages are read to initialize. Not an urgent chance for basic functionality

### CODE REVIEW

File: src/fswAlgorithms/pointCloudProcessing/initializeICP/initializeICP.cpp
Comment (patkenneally) [inline src/fswAlgorithms/pointCloudProcessing/initializeICP/initializeICP.cpp]: :small_red_triangle_down: I'm going cross eyed, or this line is a duplicate of the one right above?

File: src/fswAlgorithms/pointCloudProcessing/initializeICP/initializeICP.h
Comment (patkenneally) [inline src/fswAlgorithms/pointCloudProcessing/initializeICP/initializeICP.h]: :small_red_triangle_down: this field is never used.

File: src/fswAlgorithms/pointCloudProcessing/initializeICP/initializeICP.h
Comment (patkenneally) [inline src/fswAlgorithms/pointCloudProcessing/initializeICP/initializeICP.h]: :small_red_triangle_down: This method is not implemented.

File: src/fswAlgorithms/pointCloudProcessing/initializeICP/_UnitTest/test_initICP.py
Comment (patkenneally) [inline src/fswAlgorithms/pointCloudProcessing/initializeICP/_UnitTest/test_initICP.py]: :large_orange_diamond: Are `glob`, `pickle`, and `pytest` unused?

File: src/fswAlgorithms/pointCloudProcessing/initializeICP/_UnitTest/test_initICP.py
Comment (patkenneally) [inline src/fswAlgorithms/pointCloudProcessing/initializeICP/_UnitTest/test_initICP.py]: :large_orange_diamond: Copy and paste? Function name could be `test_all_valid_no_normalization` or something.

File: src/fswAlgorithms/pointCloudProcessing/initializeICP/_UnitTest/test_initICP.py
Comment (patkenneally) [inline src/fswAlgorithms/pointCloudProcessing/initializeICP/_UnitTest/test_initICP.py:135]: :large_orange_diamond: If you don't want an `rtol` or `atol` check then the one you don't want should be set to 0 otherwise then `rtol` defaults to 1E-7 and `atol` to 0 (probably fine).

File: src/fswAlgorithms/pointCloudProcessing/SICP/scalingIterativeClosestPoint.h
Comment (patkenneally) [inline src/fswAlgorithms/pointCloudProcessing/SICP/scalingIterativeClosestPoint.h:42]: Nice! Some explicit goodness.

**Review by patkenneally — APPROVED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: src/fswAlgorithms/pointCloudProcessing/initializeICP/initializeICP.cpp
Comment (thibaudteil) [reply src/fswAlgorithms/pointCloudProcessing/initializeICP/initializeICP.cpp]: Not cross eyed, it is fixed now!

File: src/fswAlgorithms/pointCloudProcessing/initializeICP/initializeICP.h
Comment (thibaudteil) [reply src/fswAlgorithms/pointCloudProcessing/initializeICP/initializeICP.h]: Fixed

File: src/fswAlgorithms/pointCloudProcessing/initializeICP/initializeICP.h
Comment (thibaudteil) [reply src/fswAlgorithms/pointCloudProcessing/initializeICP/initializeICP.h]: Fixed

File: src/fswAlgorithms/pointCloudProcessing/initializeICP/_UnitTest/test_initICP.py
Comment (thibaudteil) [reply src/fswAlgorithms/pointCloudProcessing/initializeICP/_UnitTest/test_initICP.py]: Yes and they are removed now

File: src/fswAlgorithms/pointCloudProcessing/initializeICP/_UnitTest/test_initICP.py
Comment (thibaudteil) [reply src/fswAlgorithms/pointCloudProcessing/initializeICP/_UnitTest/test_initICP.py]: Done!

File: src/fswAlgorithms/pointCloudProcessing/initializeICP/_UnitTest/test_initICP.py
Comment (thibaudteil) [reply src/fswAlgorithms/pointCloudProcessing/initializeICP/_UnitTest/test_initICP.py:135]: Added for clarity

File: src/fswAlgorithms/pointCloudProcessing/SICP/scalingIterativeClosestPoint.h
Comment (thibaudteil) [reply src/fswAlgorithms/pointCloudProcessing/SICP/scalingIterativeClosestPoint.h:42]: Small improvements every day

**Pull request merged**
